pkg/kubectl/cmd/get.go

```diff
@@ -248,6 +248,22 @@ func RunGet(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string
 	sorting, err := cmd.Flags().GetString("sort-by")
 	var sorter *kubectl.RuntimeSort
 	if err == nil && len(sorting) > 0 && len(objs) > 1 {
+
+		clientConfig, err := f.ClientConfig()
+		if err != nil {
+			return err
+		}
+
+		version, err := cmdutil.OutputVersion(cmd, clientConfig.GroupVersion)
+		if err != nil {
+			return err
+		}
+
+		objs, err = resource.AsVersionedObjects(infos, version.String(), f.JSONEncoder())
+		if err != nil {
+			return err
+		}
+
 		// TODO: questionable
 		if sorter, err = kubectl.SortObjects(f.Decoder(true), objs, sorting); err != nil {
```

Comment:
at first glance, rather than converting into versions objects, here, I *think* would expect to pass the output version to `SortObjects`... it is taking a fieldInput, which is not really meaningful unless paired with a group version. In `SortObjects`, can we decode to the specified groupversion? I'd like @smarterclayton's take on that as well...